src/sklearn_compat/_sklearn_compat.py

```diff
@@ -445,6 +445,71 @@ def check_array(
             **kwargs,
         )
 
+    def check_X_y(
+        X,
+        y,
+        accept_sparse=False,
+        *,
+        accept_large_sparse=True,
+        dtype="numeric",
+        order=None,
+        copy=False,
+        force_writeable=False,
+        force_all_finite="deprecated",
+        ensure_all_finite=None,
+        ensure_2d=True,
+        allow_nd=False,
+        multi_output=False,
+        ensure_min_samples=1,
+        ensure_min_features=1,
+        y_numeric=False,
+        estimator=None,
+    ):
+        """Input validation for standard estimators.
+
+        Check the original documentation for more details:
+        https://scikit-learn.org/stable/modules/generated/sklearn.utils.check_X_y.html
+        """
+        from sklearn.utils.validation import check_X_y as _check_X_y
+
+        if ensure_all_finite is not None and force_all_finite != "deprecated":
```

Comment:
Actually, if people starts to import this function, it means that need to modify this code. Therefore, there is no point to allow them passing `force_all_finite` and tell them that it is deprecated. We can remove the parameter and we make them write the code that should be written in sklearn 1.6.

However, internally we rewired as you did below.

Reply:
Reason to allow for `force_all_finite` was to match scikit-learn 1.6 signature and behavior. This would need addressing in 1.8 (or 1.7?) when the argument will be entirely removed.

I will follow along and remove it from here - Unwanted opinion, I think it is ok to be more strict in this library

Reply:
> This would need addressing in 1.8 (or 1.7?) when the argument will be entirely removed.

Yep this is my point here. You can have this unwanted scenario:

- A developer change his code but only the import to use `sklearn_compat`. If not changing anything else, a warning will be raised.
- The only to silence the warning is to use `ensure_all_finite` in addition of the change in import.
- It therefore means that there is not use for `force_all_finite` :).

Basically, the philosophy here is to really write the code as it should be in 1.6 and later, omitting the deprecation because it is what we try to remove as a developer.